lib/schemaUtils.js

```diff
@@ -2257,7 +2260,14 @@ module.exports = {
     // When processing a reference, schema.type could also be undefined
     if (schema && schema.type) {
       if (typeof schemaTypeToJsValidator[schema.type] === 'function') {
-        if (!schemaTypeToJsValidator[schema.type](valueToUse)) {
+        let isCorrectType = schemaTypeToJsValidator[schema.type](valueToUse);
+
+        // Treat unresolved postman collection/environment variable as correct type
+        if (options.ignoreUnresolvedVariables && !isCorrectType && isPmVariable(valueToUse)) {
```

Comment:
if (options.ignoreUnresolvedVariables && isPmVariable(valueToUse)) {
 // set to true
}
else {
// set to schemaTypeToJsValidator[schema.type](valueToUse)
}